catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/MetacardIngestNetworkPlugin.java

```diff
@@ -0,0 +1,166 @@
+        this.newAttributes = newAttributes;
+    }
+
+    /**
+     * Default constructor.
+     */
+    public MetacardIngestNetworkPlugin(KeyValueParser keyValueParser,
+            AttributeHelper attributeHelper) {
+        this(keyValueParser, attributeHelper, new MetacardCondition());
+    }
+
+    /**
+     * Extended constructor with a {@link MetacardCondition}.
+     */
+    public MetacardIngestNetworkPlugin(KeyValueParser keyValueParser,
+            AttributeHelper attributeHelper, MetacardCondition metacardCondition) {
+        this.keyValueParser = keyValueParser;
+        this.attributeHelper = attributeHelper;
+        this.metacardCondition = metacardCondition;
+    }
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException {
+        apply(input.getMetacards(), getClientInfoProperties());
+        return input;
+    }
+
+    //region - Passthrough  methods
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest input,
+            Map<String, Metacard> existingMetacards) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public DeleteRequest processPreDelete(DeleteRequest input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse processPostDelete(DeleteResponse input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public QueryRequest processPreQuery(QueryRequest input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public QueryResponse processPostQuery(QueryResponse input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public ResourceRequest processPreResource(ResourceRequest input)
+            throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public ResourceResponse processPostResource(ResourceResponse input, Metacard metacard)
+            throws StopProcessingException {
+        return input;
+    }
+
+    //endregion
+
+    private void apply(List<Metacard> metacards, Map<String, Serializable> criteria) {
+        if (metacardCondition.applies(criteria)) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Metacard condition was true; expected {} of {} equals {}",
+                        metacardCondition.getCriteriaKey(),
+                        metacardCondition.getExpectedValue(),
+                        criteria.get(metacardCondition.getCriteriaKey()));
+            }
+            Map<String, String> parsedAttributes = keyValueParser.parsePairsToMap(newAttributes);
+            attributeHelper.applyNewAttributes(metacards, parsedAttributes);
+        }
+    }
+
+    private Map<String, Serializable> getClientInfoProperties() throws StopProcessingException {
+        Object info = ThreadContext.get(CLIENT_INFO_KEY);
+        if (info == null || !(info instanceof Map)) {
```

Comment:
Correct that null info will never be an instance of map, but we can't remove the map check. We expect that client-info takes the form of a map (see the client-info-plugin) because we are performing a cast and treating it as a map.  Suppose `info != null`, then we move on to the 2nd check.